### ISSUE TITLE
OAuth2 tokenURL is relative to server domain only and ignores server URL path

### ISSUE THREAD

**Issue by ShadowLNC** [rapi-doc/RapiDoc#1046]
Per #1037 opening a new issue as requested

Given:

* A server URL `https://example.com/api`, and
* An OAuth2 token URL `/subpath/token`

Then the resulting URL should be `https://example.com/api/subpath/token`.

However, with RapiDoc, the `/api` part is missing. The same examples apply as per #1037 including the screenshot provided by @mrin9

**Comment by mrin9** [comment]
> Then the resulting URL should be https://example.com/api/subpath/token

Where are you getting this from, provide me with a source ? per the OpenAPI 3.0+ Specs it should be `https://example.com/subpath/token`

And here is the source of how [relative endpoints are defined](https://swagger.io/docs/specification/v3_0/authentication/oauth2/#relative-endpoint-urls) in Open API

Also you can tru the swagger [online editor](https://editor-next.swagger.io/) and provide ur spec and click Authorize you will notice the API call is made to `https://example.com/subpath/token` without `api` part in it

**Comment by ShadowLNC** [comment]
So I read through the spec you linked and it references [RFC3986 (specifically §4.2 which in turn references §5](https://datatracker.ietf.org/doc/html/rfc3986#section-5.2.2). This seems to indicate that if the leading slash is removed, it should allow the URL to be merged rather than replaced.

I tried with a server URL ending in `api/` (trailing slash) and the token URL of `subpath/token` (removed leading slash).

* In RapiDoc with these updated URLs, I got the same result (missing `/api`)
* In the Swagger editor, it does include `/api`
* In the Swagger editor, the original example URLs result in missing `/api` (appears to be consistent with the spec)

Apologies for the confusion - I did try these combinations originally, but since they did not behave differently in RapiDoc, I didn't realise the spec actually declares that they should be handled differently. I think I got confused because my linter bans trailing slashes from server URLs, and endpoint definitions have to include a leading slash.
